### PR TITLE
Trezor GPG Agent

### PR DESCRIPTION
**Pull request by jb55**
###### Motivation for this change

This adds an option to enable trezor-agent, a GPG agent that enables GPG encrypting and signing with a Trezor hardware wallet.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/top-level/python-packages.nix [hunk 1]
```diff
@@ -22388,18 +22388,27 @@ EOF
   trezor_agent = buildPythonPackage rec{
```
Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: We are moving Python expressions out of `pkgs/top-level/python-packages.nix` into `pkgs/development/python-modules/<module>/default.nix`. 

Please move the expression there, and call it from `pkgs/top-level/python-packages.nix` using `callPackage ../development/python-modules/<package> { };`.

**Review by FRidh — CHANGES_REQUESTED** [review]

**Comment by FRidh** [comment]
@fpletz is it ok to add trezor to this module?

File: pkgs/top-level/python-packages.nix (hunk 1 above)
Comment (jb55) [reply pkgs/top-level/python-packages.nix]: thanks for reviewing this! will do

**Review by fpletz — APPROVED** [review]
👍 on putting it in the gnupg module but please move the python module as @FRidh suggested.

**Comment by jb55** [comment]
libagent keeps breaking, this time due to a few deps. bah! will have to revisit.

**Pull request closed without merging**